mlflow/tracing/export/mlflow.py

```diff
@@ -84,18 +84,21 @@ def _export_trace(self, root_span: MLflowSpanWrapper):
         # TODO: Make this async
         self._client.log_trace(trace)
 
-    def _serialize_inputs_outputs(self, input_or_output: Optional[Dict[str, Any]]) -> str:
+    def _serialize_inputs_outputs(self, input_or_output: Optional[Any]) -> str:
         """
         Serialize inputs or outputs field of the span to a string, and truncate if necessary.
         """
         if not input_or_output:
             return ""
 
         try:
-            serialized = json.dumps(input_or_output)
+            serialized = json.dumps(input_or_output, default=str)
         except TypeError:
-            # If not JSON-serializable, use string representation
-            serialized = str(input_or_output)
+            # If not JSON/string serializable, raise a warning and return an empty string
+            _logger.warning(
+                "Failed to serialize inputs/outputs for a trace, an empty string will be recorded."
+            )
+            return ""
```

Comment:
We probably won't reach here unless you have a class like this:

```python
import json


class DoNotStrMe:
    def __str__(self) -> str:
        raise TypeError("I hate being str-ed")


json.dumps(DoNotStrMe(), default=str)
```

Reply:
yeah but this is the block for such weird class:)